src/py/flwr/server/superlink/fleet/vce/backend/raybackend.py

```diff
@@ -0,0 +1,152 @@
+
+from flwr.client.clientapp import ClientApp
+from flwr.common.context import Context
+from flwr.common.logger import log
+from flwr.common.message import Message
+from flwr.simulation.ray_transport.ray_actor import (
+    BasicActorPool,
+    ClientAppActor,
+    init_ray,
+)
+
+from .backend import Backend, BackendConfig
+
+ClienteResourcesDict = Dict[str, Union[int, float]]
+
+
+class RayBackend(Backend):
+    """A backend that submits jobs to a `BasicActorPool`."""
+
+    def __init__(
+        self,
+        backend_config: BackendConfig,
+        work_dir: str,
+    ) -> None:
+        """Prepare RayBackend by initialising Ray and creating the ActorPool."""
+        log(INFO, "Initialising: %s", self.__class__.__name__)
+        log(INFO, "Backend config: %s", backend_config)
+
+        # Init ray and append working dir if needed
+        runtime_env = (
+            self._configure_runtime_env(work_dir=work_dir) if work_dir else None
+        )
+        init_ray(runtime_env=runtime_env)
+
+        # Validate client resources
+        self.client_resources_key = "client_resources"
+
+        # Create actor pool
+        actor_kwargs = backend_config.get("actor_kwargs", {})
+        client_resources = self._validate_client_resources(config=backend_config)
+        self.pool = BasicActorPool(
+            actor_type=ClientAppActor,
+            client_resources=client_resources,
+            actor_kwargs=actor_kwargs,
+        )
+
+    def _configure_runtime_env(self, work_dir: str) -> Dict[str, Union[str, List[str]]]:
+        """Return list of files/subdirectories to exclude relateive to work_dir.
+
+        Without this, Ray will push everything to the Ray Cluster.
+        """
+        runtime_env: Dict[str, Union[str, List[str]]] = {"working_dir": work_dir}
+
+        if runtime_env:
+            excludes = []
+            path = pathlib.Path(work_dir)
+            for p in path.rglob("*"):
+                # exclude files need to be relative to the working_dir
+                if p.is_file() and not str(p).endswith(".py"):
+                    excludes.append(str(p.relative_to(path)))
+            runtime_env["excludes"] = excludes
+
+        return runtime_env
+
+    def _validate_client_resources(self, config: BackendConfig) -> ClienteResourcesDict:
+        client_resources_config = config.get(self.client_resources_key)
+        client_resources: ClienteResourcesDict = {}
+        valid_types = (int, float)
+        if client_resources_config:
+            for k, v in client_resources_config.items():
+                assert isinstance(k, str), ValueError(
+                    f"client resources keys are expected to be `str` but you used "
+                    f"{type(k)} for `{k}`"
+                )
+                assert isinstance(v, valid_types), ValueError(
+                    f"client resources are expected to be of type {valid_types} but "
+                    f"found `{type(v)}` for key `{k}`",
+                )
```

Comment:
We usually don't have `assert` in the production part of the codebase, only in tests.